src/widgets/barchart.rs

```diff
@@ -335,6 +347,32 @@ impl<'a> BarChart<'a> {
     }
 
     fn render_horizontal_bars(self, buf: &mut Buffer, bars_area: Rect, max: u64) {
+        let label_size = if let Some(length) = self.label_max_len {
+            length
+        } else {
+            // get the longest label
+            self.data
+                .iter()
+                .map(|group| {
+                    group
+                        .bars
+                        .iter()
+                        .map(|bar| bar.label.clone().map_or(0, |label| label.width()))
+                        .max()
+                        .unwrap_or(0)
+                })
+                .max()
+                .unwrap_or(0) as u16
+        };
+
+        let margin = if label_size == 0 { 0 } else { 1 };
+
+        let bars_area = Rect {
+            x: bars_area.x + label_size + margin,
```

Comment:
I would save `bar_area.x` to use it later for the labels instead of calculating it again inside for loops.
`let label_x = bars_area.x;`